be1-go/hub/standard_hub/mod.go

```diff
@@ -79,13 +85,21 @@ type Hub struct {
 	// rootInbox and queries are used to help servers catchup to each other
 	rootInbox inbox.Inbox
 	queries   queries
+
+	//globalInboy is used to remember all messages
+	globalInbox inbox.Inbox
+
+	// messageIdsByChannel stores all the message ids and the corresponding channel ids
+	// to help servers
```

Comment:
To help servers determine which channel corresponds to the message ids since the channel isn't stored in the message. I'll clarify that in a new commit.